Bumping up background sweep logs priority

fsamuel-bs opened the pull request:
**Goals (and why)**: With the rollout of background sweep, we should bump sweep priority to better perceive the effects of sweeping.

**Implementation Description (bullets)**:
- Bump logs to info and warn accordingly.
- Mark arguments as safe or unsafe accordingly.

**Concerns (what feedback would you like?)**: 
- Am I logging something unsafe?
- Did I forget any important logs?
- Am I logging to much?

**Where should we start reviewing?**: -

**Priority (whenever / two weeks / yesterday)**: Since it's small, EOW

<!---
Please remember to:
- Add any necessary release notes (including breaking changes)
- Make sure the documentation is up to date for your change
--->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/palantir/atlasdb/2031)
<!-- Reviewable:end -->

gmaretic commented on atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java:
```diff
@@ -234,11 +238,15 @@ private void runOnceForTable(TableToSweep tableToSweep) {
                             + " up to timestamp {}.",
-                    results.getCellTsPairsExamined(), tableRef, startRowToHex(startRow),
-                    results.getStaleValuesDeleted(), elapsedMillis, results.getSweptTimestamp());
+                    SafeArg.of("", results.getCellTsPairsExamined()),
```
This looks unsafe

fsamuel-bs replied:
This seems to be the number of cells examined (check the log message as well line 352 of the same file, which is where we set the property). I would say it's safe, but maybe I'm missing something.

hsaraogi replied:
`SafeArg.of("cellTsPairsExamined", results.getCellTsPairsExamined())`. In general please dont skip the parameter string, in the future internal logging product will use them for analysing logs.

jeremyk-91 commented on atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java:
```diff
@@ -234,11 +238,15 @@ private void runOnceForTable(TableToSweep tableToSweep) {
-                    results.getStaleValuesDeleted(), elapsedMillis, results.getSweptTimestamp());
+                    SafeArg.of("", results.getCellTsPairsExamined()),
+                    UnsafeArg.of("table name", tableRef),
+                    SafeArg.of("row hex", startRowToHex(startRow)),
```
This can be decoded to recover the row key, so should be unsafe.

gmaretic replied:
IIRC, this is not necessarily a row key, but the old way paging worked would cause this value to be the next byte array value that follows the value of the last row from a previous page. Hadn't looked into the new sweep code, but probably best to err on the side of caution and make it unsafe.

fsamuel-bs replied:
Agreed, making it unsafe.

jeremyk-91 commented on atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java:
```diff
@@ -234,11 +238,15 @@ private void runOnceForTable(TableToSweep tableToSweep) {
+                    SafeArg.of("row hex", startRowToHex(startRow)),
+                    SafeArg.of("", results.getStaleValuesDeleted()),
+                    SafeArg.of("", elapsedMillis),
+                    SafeArg.of("", results.getSweptTimestamp()));
```
Let's try to include names for these logging args (if that's not too much work)

nziebart replied:
per internal discussions, we're leaning towards not doing substitution, so names are critical

jeremyk-91 commented on atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java:
```diff
@@ -248,10 +256,10 @@ private void runOnceForTable(TableToSweep tableToSweep) {
+            log.warn("Failed to sweep {} with batch config {} starting from row {}",
+                    UnsafeArg.of("table name", tableRef),
+                    SafeArg.of("", batchConfig),
+                    SafeArg.of("hex", startRowToHex(startRow)));
```
As above, if I'm not wrong this is a row key

jeremyk-91 commented on atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java:
```diff
@@ -197,7 +199,9 @@ long grabLocksAndRun(SweepLocks locks) throws InterruptedException {
             } else {
                 SweepBatchConfig lastBatchConfig = getAdjustedBatchConfig();
                 log.error("The background sweep job failed unexpectedly with batch config {}"
```
We should discuss this (and I realise not the scope of your PR) but `ERROR` feels excessive here. I'd prefer `WARN` if >1 and `ERROR` only if =1. (Don't block on this comment!)

fsamuel-bs replied:
Agreed. Downgrading to warn here.

Should I also downgrade on the logging above? I would say it's not an error, as in, sweep will recover properly from that situation. I would probably downgrade that to info.

hsaraogi commented on atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java:
```diff
@@ -183,10 +185,10 @@ long grabLocksAndRun(SweepLocks locks) throws InterruptedException {
                 }
             } else {
-                log.debug("Skipping sweep because it is currently disabled.");
+                log.info("Skipping sweep because it is currently disabled.");
```
This will keep spamming the logs if the config flag is disabled. Will leave it at debug.

hsaraogi commented on atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java:
```diff
@@ -183,10 +185,10 @@ long grabLocksAndRun(SweepLocks locks) throws InterruptedException {
                     sweptSuccessfully = runOnce();
                 } else {
-                    log.debug("Skipping sweep because sweep is running elsewhere.");
+                    log.warn("Skipping sweep because sweep is running elsewhere.");
```
Not really a warn, something is being swept.. debug seems fine.

hsaraogi commented on atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java:
```diff
@@ -197,7 +199,9 @@ long grabLocksAndRun(SweepLocks locks) throws InterruptedException {
                 log.error("The background sweep job failed unexpectedly with batch config {}"
-                        + ". Attempting to continue with a lower batch size...", lastBatchConfig, e);
+                        + ". Attempting to continue with a lower batch size...",
+                        SafeArg.of("", lastBatchConfig),
```
`SafeArg.of("lastBatchConfig", lastBatchConfig);`

hsaraogi commented on atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java:
```diff
@@ -322,7 +330,7 @@ OptionalLong getPreviousMinimumSweptTimestamp() {
                             tx, sweepRunner.getConservativeSweepTimestamp());
                     if (nextTable.isPresent()) {
-                        log.debug("Now starting to sweep {}.", nextTable);
+                        log.info("Now starting to sweep {}.", UnsafeArg.of("next table", nextTable));
```
Leave at debug especially as table is not collected.. finished sweeping info is enough to confirm sweep has been running

hsaraogi commented on atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java:
```diff
@@ -361,7 +371,9 @@ private void performInternalCompactionIfNecessary(TableReference tableRef, Sweep
             kvs.compactInternally(tableRef);
             long elapsedMillis = watch.elapsed(TimeUnit.MILLISECONDS);
-            log.debug("Finished performing compactInternally on {} in {} ms.", tableRef, elapsedMillis);
+            log.info("Finished performing compactInternally on {} in {} ms.",
```
Leave at debug, compaction is know to not work, so no useful info here.. @ashrayjain for SA.

hsaraogi commented on atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java:
```diff
@@ -446,7 +458,7 @@ public synchronized void shutdown() {
             return;
         }
-        log.debug("Signalling background sweeper to shut down.");
+        log.info("Signalling background sweeper to shut down.");
```
This will be logged at WARN `log.warn("Shutting down background sweeper. Please restart the service to rerun background sweep.");`.. this will just be spammy.

fsamuel-bs commented:
I guess I've addressed all the comments on this pr. All the logs now have corresponding keys, and the priorities have been reassigned as requested. It's ready for review again.

fsamuel-bs commented on atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java:
```diff
@@ -193,11 +195,13 @@ long grabLocksAndRun(SweepLocks locks) throws InterruptedException {
             sweepMetrics.sweepError();
             if (checkAndRepairTableDrop()) {
-                log.error("The table being swept by the background sweeper was dropped, moving on...");
+                log.info("The table being swept by the background sweeper was dropped, moving on...");
```
I would say this line is not an error, as sweep will recover properly from that situation, so I downgraded to info. Is my reasoning correct?

gmaretic replied:
You could log it as a warning, but info is fine

fsamuel-bs commented:
Resolve #1565.
